BridgeSDK/BridgeAPI/SBBUploadManager.m

```diff
@@ -286,26 +291,36 @@ - (void)toldBridgeFileUploadedWithTask:(NSURLSessionUploadTask *)uploadTask forB
 - (void)uploadedFileToS3WithTask:(NSURLSessionUploadTask *)uploadTask response:(NSHTTPURLResponse *)httpResponse error:(NSError *)error
 {
     NSInteger httpStatusCode = httpResponse.statusCode;
+    NSString *filePath = uploadTask.taskDescription;
     
     // client-side networking issue
     if (error) {
-        [self completeUploadOfFile:uploadTask.taskDescription withError:error];
+        [self completeUploadOfFile:filePath withError:error];
         return;
     }
     
     // server didn't like the request, or otherwise hiccupped
-    if (httpStatusCode >= 300) {
+    if (httpStatusCode == 403) {
+        // this will have happened because the pre-signed url timed out before starting the actual upload to S3,
+        // so try, try again
+#if DEBUG
+        NSLog(@"Background file upload to S3 failed with HTTP status 403 (presumably due to presigned url timeout); retrying...");
+#endif
+        NSDictionary *uploadRequestJSON = [self uploadRequestJSONForFile:filePath];
+        [self kickOffUploadForFile:filePath uploadRequestJSON:uploadRequestJSON];
+        return;
+    } else if (httpStatusCode >= 300) {
         // iOS handles redirects automatically so only e.g. 307 resource not changed etc. from the 300 range should end up here
         // (along with all 4xx and 5xx of course)
         NSString *description = [NSString stringWithFormat:@"Background file upload to S3 failed with HTTP status %ld", (long)httpStatusCode];
         NSError *s3Error = [NSError errorWithDomain:SBB_ERROR_DOMAIN code:SBBErrorCodeS3UploadErrorResponse userInfo:@{NSLocalizedDescriptionKey: description}];
-        [self completeUploadOfFile:uploadTask.taskDescription withError:s3Error];
+        [self completeUploadOfFile:filePath withError:s3Error];
         return;
     }
```

Comment:
This is the kind of a case where I find nested if statements and explicit special-case handling a little easier to read.

````
    // server didn't like the request, or otherwise hiccupped
    if (httpStatusCode >= 300) {
        switch (httpStatusCode) {
            case 403:{
#if DEBUG
                NSLog(@"Background file upload to S3 failed with HTTP status 403 (presumably due to presigned url timeout); retrying...");
#endif
                // this will have happened because the pre-signed url timed out before starting the actual upload to S3,
                // so try, try again
                NSDictionary *uploadRequestJSON = [self uploadRequestJSONForFile:filePath];
                [self kickOffUploadForFile:filePath uploadRequestJSON:uploadRequestJSON];
            } break;
                
            default:{
                // iOS handles redirects automatically so only e.g. 307 resource not changed etc. from the 300 range should end up here
                // (along with all 4xx and 5xx of course)
                NSString *description = [NSString stringWithFormat:@"Background file upload to S3 failed with HTTP status %ld", (long)httpStatusCode];
                NSError *s3Error = [NSError errorWithDomain:SBB_ERROR_DOMAIN code:SBBErrorCodeS3UploadErrorResponse userInfo:@{NSLocalizedDescriptionKey: description}];
                [self completeUploadOfFile:filePath withError:s3Error];
            } break;
        }

        // exit early
        return;
    }
````

Possibly moving the entire switch statement into an extracted method for readability?